deploy/crds/infinispan.org_backups_crd.yaml

```diff
@@ -0,0 +1,1285 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: backups.infinispan.org
+  labels:
+    name: infinispan-operator
+spec:
+  group: infinispan.org
+  names:
+    kind: Backup
+    listKind: BackupList
+    plural: backups
+    singular: backup
+  scope: Namespaced
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      description: Backup is the Schema for the backups API
+      properties:
+        apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+          type: string
+        kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          type: object
+        spec:
+          description: BackupSpec defines the desired state of Backup
+          properties:
+            cluster:
+              type: string
+            container:
+              description: InfinispanContainerSpec specify resource requirements per
+                container
+              properties:
+                cpu:
+                  type: string
+                extraJvmOpts:
+                  type: string
+                memory:
+                  type: string
+              type: object
+            resources:
+              properties:
+                cacheConfigs:
+                  items:
+                    type: string
+                  type: array
+                caches:
+                  items:
+                    type: string
+                  type: array
+                counters:
+                  items:
+                    type: string
+                  type: array
+                protoSchemas:
+                  items:
+                    type: string
+                  type: array
+                scripts:
+                  items:
+                    type: string
+                  type: array
+              type: object
+            volume:
+              properties:
+                subPath:
+                  description: Path within the mounted volume from which the container's
+                    volume should be mounted.
+                  type: string
+                volumeSource:
+                  description: The VolumeSource to utilise on the zero-capacity pod
+                  properties:
```

Comment:
Are all off this properties valid or is this copied from somewhere?

Reply:
This is generated. It is what I was refering to in the PR description, whether we can avoid all of this being dragged in.

Reply:
@ryanemerson I don't know if it's possible to switch off doc. I've looked around the code generators but found nothing.
You can have a better look here:
https://github.com/kubernetes/code-generator
https://book.kubebuilder.io/reference/generating-crd.html

Does the user need to configure all the InfinispanContainerSpec fields?

Reply:
> Does the user need to configure all the InfinispanContainerSpec fields?

No, these will be optional. I thought it would be useful to have in the Spec if a user wanted to provision a more powerful pod to perform a backup/restore.

Reply:
imo it's ok to keep the documentation if the user can provide custom configuration then

Reply:
`VolumeSource` is no longer part of the backup or restore spec,  so this is no longer an issue.